### DISCUSSION TITLE
Supabase integration

### DISCUSSION THREAD

**Discussion by hidech** [webstudio-is/webstudio-community#57] [Ideas]
It will be wonderful if we can use the data I store in Supabase on WebStudio.
Supabase JS client is quite handy to fetch data.

This is a part of sample code.
If Supabase JS client can be loaded by WebStudio, it is more efficient.

```
<script src="https://cdn.jsdelivr.net/npm/@supabase/supabase-js@2"></script>
<script>
  window.addEventListener('load', () => {
    if (window.supabase) {
      let supabase = window.supabase.createClient('URL','Anon-key');
     async function fetchData() {
        const { data, error } = await supabase.from('posts').select("title, summary");
...
```

**Comment by kof** [comment]
Absolutely, also you would want to be able to run data fetching serverside, so that when you serve the page, its already populated with data and doesn't start data fetching after client has loaded

**Reply by hidech** [reply]
Supabase's server-side rendering is steadily being implemented. It will be difficult, but if this is integrated, we will have a wonderful future.
https://supabase.com/docs/guides/auth/server-side-rendering

**Reply by kof** [reply]
we have our own SSR already, just currently no way to fetch data serverside by the user, this is needed for all data integrations

**Reply by hidech** [reply]
Thank you for your reply. I hope I can get my hands on it someday.

**Comment by TrySound** [comment]
We landed data bindings. Here's example of supabase integration

https://www.youtube.com/watch?v=2MrkfnjYuCo

**Reply by hidech** [reply]
Amazing! Supabase integration should be game changer.
